openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -39,9 +39,16 @@ def process_request(self, request):
             preferred_language = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
             lang_headers = [seq[0] for seq in parse_accept_lang_header(preferred_language)]
 
+            prefixes = [prefix.split("-")[0] for prefix in system_released_languages]
```

Comment:
Outside the scope of this PR: 
It looks like this middleware is doing the same string parsings for every request (call to `released_languages` and now this new code to parse `prefixes`).  This is something we should compute only once in-process and "memoize" for future requests.